### PR TITLE
MOTS-115: Expiring session

### PR DESCRIPTION
**Pull request by dpiaseckisoldevelo**

### CODE REVIEW

File: src/webapp/js/reducers/auth_reducer.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
-import { AUTH_USER, UNAUTH_USER, AUTH_ERROR } from '../actions/types';
+import { AUTH_USER, UNAUTH_USER, AUTH_ERROR, SET_COUNTER_LOGOUT_TIME } from '../actions/types';
 
-export default function (state = { error: null, authenticated: false }, action) {
+const initialState = {
+  error: null,
+  authenticated: false,
+  counterLogoutTime: 20,
```
Comment (pmuchowski) [inline src/webapp/js/reducers/auth_reducer.js]: I think this should be 600 (10 minutes)

File: src/webapp/js/reducers/auth_reducer.js [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+export default function (state = initialState, action) {
   switch (action.type) {
     case AUTH_USER:
       return { ...state, error: null, authenticated: true };
     case UNAUTH_USER:
       localStorage.removeItem('token');
       localStorage.removeItem('refresh_token');
       return { ...state, authenticated: false };
+    case SET_COUNTER_LOGOUT_TIME:
+      return { ...state, counterLogoutTime: action.payload, resetCounter: !state.resetCounter };
```
Comment (pmuchowski) [inline src/webapp/js/reducers/auth_reducer.js]: Why resetCounter in not set to true?

File: src/webapp/js/components/counter-logout.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+export default connect(mapStateToProps, { signoutUser })(CounterLogout);
+
+CounterLogout.propTypes = {
+  counterLogoutTime: PropTypes.number,
+  signoutUser: PropTypes.func.isRequired,
+};
+
+CounterLogout.defaultProps = {
+  counterLogoutTime: 300,
```
Comment (pmuchowski) [inline src/webapp/js/components/counter-logout.js]: I think this should be set to 600 (10 minutes)

Comment (dpiaseckisoldevelo) [reply src/webapp/js/components/counter-logout.js]: Make sense, I changed this values for debug purposes.

**Pull request merged**